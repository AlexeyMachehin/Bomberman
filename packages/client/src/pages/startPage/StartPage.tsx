import NavigateLinks from '../../features/startPage/components/navigateLinks/NavigateLinks';
import NumberOfPlayersButtons from '../../features/startPage/components/numberOfPlayersButtons/NumberOfPlayersButtons';
import { ThemeProvider, createTheme } from '@mui/material/styles';
import { Typography } from '@mui/material';
import classes from './startPage.module.css';

const theme = createTheme({
  typography: {
    fontFamily: 'Orbitron',
  },
});

export default function StartPage() {
  return (
    <ThemeProvider theme={theme}>
      <div data-testid="startPage-component" className={classes.startPageWrapper}>
        <Typography variant="h2" className={classes.startPageTitle}>
          BOMBERMAN
        </Typography>
        <NumberOfPlayersButtons />
        <NavigateLinks />
      </div>
    </ThemeProvider>
  );
}
