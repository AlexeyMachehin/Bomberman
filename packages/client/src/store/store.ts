import { configureStore } from '@reduxjs/toolkit';
import { userSlice } from './user/userSlice';
import { forumSlice } from './forum/forumSlice';

export const createStore = () => {
  return configureStore({
    reducer: {
      userReducer: userSlice.reducer,
      forumReducer: forumSlice.reducer,
    },
  });
};

export const store = createStore();

// Infer the `RootState` and `AppDispatch` types from the store itself
export type RootState = ReturnType<typeof store.getState>;
// Inferred type: {posts: PostsState, comments: CommentsState, users: UsersState}
export type AppDispatch = typeof store.dispatch;
