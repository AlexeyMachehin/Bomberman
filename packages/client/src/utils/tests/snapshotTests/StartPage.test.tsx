import '../../../matchMedia.mock';
import StartPage from '@/pages/startPage/StartPage';
import { render } from '@testing-library/react';
import { store } from '@/store/store';
import { Provider } from 'react-redux';
import { MemoryRouter, Route, Routes } from 'react-router-dom';
import { Route as RoutePath } from '@/const';

test('StartPage renders correctly', () => {
  const component = render(
    <MemoryRouter>
      <Provider store={store}>
        <Routes>
          <Route path={RoutePath.INDEX} element={<StartPage />} />
        </Routes>
      </Provider>
    </MemoryRouter>
  );
  expect(component).toMatchSnapshot();
});
