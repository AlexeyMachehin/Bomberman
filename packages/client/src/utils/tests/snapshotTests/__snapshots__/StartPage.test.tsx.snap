// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StartPage renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <span
          class="MuiSwitch-root MuiSwitch-sizeMedium css-lu9khh-MuiSwitch-root"
        >
          <span
            class="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary Mui-checked PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary Mui-checked Mui-checked css-5ryogn-MuiButtonBase-root-MuiSwitch-switchBase"
          >
            <input
              checked=""
              class="PrivateSwitchBase-input MuiSwitch-input css-1m9pwf3"
              type="checkbox"
            />
            <span
              class="MuiSwitch-thumb css-jsexje-MuiSwitch-thumb"
            />
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </span>
          <span
            class="MuiSwitch-track css-1yjjitx-MuiSwitch-track"
          />
        </span>
      </div>
      <div
        data-testid="startPage-component"
      >
        <h2
          class="MuiTypography-root MuiTypography-h2 css-1hcdupa-MuiTypography-root"
        >
          BOMBERMAN
        </h2>
        <div
          class="MuiBox-root css-0"
        >
          <button
            class="MuiButtonBase-root css-5nngyh-MuiButtonBase-root"
            style="width: 50%;"
            tabindex="0"
            type="button"
          >
            <span
              class="undefined undefined"
            />
            <span
              class="MuiImageBackdrop-root undefined"
            />
            <span>
              <span
                class="MuiTypography-root MuiTypography-body1 css-1b2aw8c-MuiTypography-root"
              >
                ONE PLAYER
                <span
                  class="MuiImageBackdrop-root undefined"
                />
              </span>
            </span>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButtonBase-root css-5nngyh-MuiButtonBase-root"
            style="width: 50%;"
            tabindex="0"
            type="button"
          >
            <span
              class="undefined undefined"
            />
            <span
              class="MuiImageBackdrop-root undefined"
            />
            <span>
              <span
                class="MuiTypography-root MuiTypography-body1 css-1b2aw8c-MuiTypography-root"
              >
                TWO PLAYERS
                <span
                  class="MuiImageBackdrop-root undefined"
                />
              </span>
            </span>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="MuiBox-root css-kbj29x"
        >
          <a
            aria-label="Logout from system"
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-x3euig-MuiButtonBase-root-MuiButton-root"
            data-mui-internal-clone-element="true"
            href="#text-buttons"
            tabindex="0"
          >
            Logout
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </a>
          <button
            aria-label="Go to Profile/Change profile data"
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-x3euig-MuiButtonBase-root-MuiButton-root"
            data-mui-internal-clone-element="true"
            tabindex="0"
            type="button"
          >
            Profile
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            aria-label="Go to Forum"
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-x3euig-MuiButtonBase-root-MuiButton-root"
            data-mui-internal-clone-element="true"
            tabindex="0"
            type="button"
          >
            Forum
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            aria-label="Go to Leaderboard"
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-x3euig-MuiButtonBase-root-MuiButton-root"
            data-mui-internal-clone-element="true"
            tabindex="0"
            type="button"
          >
            Leaderboard
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            aria-label="Open guide"
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-x3euig-MuiButtonBase-root-MuiButton-root"
            data-mui-internal-clone-element="true"
            tabindex="0"
            type="button"
          >
            How to play
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            aria-label="Open guide"
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-x3euig-MuiButtonBase-root-MuiButton-root"
            data-mui-internal-clone-element="true"
            tabindex="0"
            type="button"
          >
            About game
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <div>
            <div>
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-1ijcipm-MuiButtonBase-root-MuiButton-root"
                style="width: 120px; margin-bottom: 10px;"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="VolumeOffIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M16.5 12c0-1.77-1.02-3.29-2.5-4.03v2.21l2.45 2.45c.03-.2.05-.41.05-.63zm2.5 0c0 .94-.2 1.82-.54 2.64l1.51 1.51C20.63 14.91 21 13.5 21 12c0-4.28-2.99-7.86-7-8.77v2.06c2.89.86 5 3.54 5 6.71zM4.27 3 3 4.27 7.73 9H3v6h4l5 5v-6.73l4.25 4.25c-.67.52-1.42.93-2.25 1.18v2.06c1.38-.31 2.63-.95 3.69-1.81L19.73 21 21 19.73l-9-9L4.27 3zM12 4 9.91 6.09 12 8.18V4z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
              <input
                max="1"
                min="0"
                step="0.05"
                type="range"
                value="1"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <span
        class="MuiSwitch-root MuiSwitch-sizeMedium css-lu9khh-MuiSwitch-root"
      >
        <span
          class="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary Mui-checked PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary Mui-checked Mui-checked css-5ryogn-MuiButtonBase-root-MuiSwitch-switchBase"
        >
          <input
            checked=""
            class="PrivateSwitchBase-input MuiSwitch-input css-1m9pwf3"
            type="checkbox"
          />
          <span
            class="MuiSwitch-thumb css-jsexje-MuiSwitch-thumb"
          />
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </span>
        <span
          class="MuiSwitch-track css-1yjjitx-MuiSwitch-track"
        />
      </span>
    </div>
    <div
      data-testid="startPage-component"
    >
      <h2
        class="MuiTypography-root MuiTypography-h2 css-1hcdupa-MuiTypography-root"
      >
        BOMBERMAN
      </h2>
      <div
        class="MuiBox-root css-0"
      >
        <button
          class="MuiButtonBase-root css-5nngyh-MuiButtonBase-root"
          style="width: 50%;"
          tabindex="0"
          type="button"
        >
          <span
            class="undefined undefined"
          />
          <span
            class="MuiImageBackdrop-root undefined"
          />
          <span>
            <span
              class="MuiTypography-root MuiTypography-body1 css-1b2aw8c-MuiTypography-root"
            >
              ONE PLAYER
              <span
                class="MuiImageBackdrop-root undefined"
              />
            </span>
          </span>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButtonBase-root css-5nngyh-MuiButtonBase-root"
          style="width: 50%;"
          tabindex="0"
          type="button"
        >
          <span
            class="undefined undefined"
          />
          <span
            class="MuiImageBackdrop-root undefined"
          />
          <span>
            <span
              class="MuiTypography-root MuiTypography-body1 css-1b2aw8c-MuiTypography-root"
            >
              TWO PLAYERS
              <span
                class="MuiImageBackdrop-root undefined"
              />
            </span>
          </span>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        class="MuiBox-root css-kbj29x"
      >
        <a
          aria-label="Logout from system"
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-x3euig-MuiButtonBase-root-MuiButton-root"
          data-mui-internal-clone-element="true"
          href="#text-buttons"
          tabindex="0"
        >
          Logout
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </a>
        <button
          aria-label="Go to Profile/Change profile data"
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-x3euig-MuiButtonBase-root-MuiButton-root"
          data-mui-internal-clone-element="true"
          tabindex="0"
          type="button"
        >
          Profile
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <button
          aria-label="Go to Forum"
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-x3euig-MuiButtonBase-root-MuiButton-root"
          data-mui-internal-clone-element="true"
          tabindex="0"
          type="button"
        >
          Forum
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <button
          aria-label="Go to Leaderboard"
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-x3euig-MuiButtonBase-root-MuiButton-root"
          data-mui-internal-clone-element="true"
          tabindex="0"
          type="button"
        >
          Leaderboard
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <button
          aria-label="Open guide"
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-x3euig-MuiButtonBase-root-MuiButton-root"
          data-mui-internal-clone-element="true"
          tabindex="0"
          type="button"
        >
          How to play
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <button
          aria-label="Open guide"
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-x3euig-MuiButtonBase-root-MuiButton-root"
          data-mui-internal-clone-element="true"
          tabindex="0"
          type="button"
        >
          About game
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <div>
          <div>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-1ijcipm-MuiButtonBase-root-MuiButton-root"
              style="width: 120px; margin-bottom: 10px;"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="VolumeOffIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M16.5 12c0-1.77-1.02-3.29-2.5-4.03v2.21l2.45 2.45c.03-.2.05-.41.05-.63zm2.5 0c0 .94-.2 1.82-.54 2.64l1.51 1.51C20.63 14.91 21 13.5 21 12c0-4.28-2.99-7.86-7-8.77v2.06c2.89.86 5 3.54 5 6.71zM4.27 3 3 4.27 7.73 9H3v6h4l5 5v-6.73l4.25 4.25c-.67.52-1.42.93-2.25 1.18v2.06c1.38-.31 2.63-.95 3.69-1.81L19.73 21 21 19.73l-9-9L4.27 3zM12 4 9.91 6.09 12 8.18V4z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <input
              max="1"
              min="0"
              step="0.05"
              type="range"
              value="1"
            />
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
